### PR TITLE
Set up Flake8 and update GitHub workflows for Python version and formatting tools

### PR DESCRIPTION
**Pull request by BruhLemma-Yadecha**
This pull request introduces Flake8 configuration and updates the Python version across GitHub workflows, ensuring consistency and improved code quality checks. Below are the key changes made:

### New Files
- **`.flake8`**: Added a configuration file for Flake8 with the following settings:
  - `max-line-length`: Set to 100 characters.
  - `ignore`: Excluded specific error codes: E203, E402, F403, F405, W503.
  - `exclude`: Excluded the following directories from checks: `.git`, `__pycache__`, `venv`, `api/migrations/`.

- **`pyproject.toml`**: Introduced a new configuration file for Black with the following settings:
  - `line-length`: Set to 100 characters.
  - `target-version`: Set to `['py312']` to align with the Python 3.12.7 version.

### Updates to GitHub Workflows
- **Flake8 Format Workflow (`.github/workflows/flake8-format.yml`)**:
  - Updated `setup-python` action from version `v2` to `v5`.
  - Upgraded Python version from `3.10.5` to `3.12.7`.
  - Simplified the Flake8 command to just `flake8`, utilizing the new `.flake8` configuration settings.

- **Python Black Check Workflow (`.github/workflows/python-black-check.yml`)**:
  - Updated `setup-python` action from version `v2` to `v5`.
  - Upgraded Python version from `3.x` to `3.12.7`.
  - Updated the Black command to use the settings specified in `pyproject.toml` by removing the `--line-length` option.

These changes enhance code consistency and help maintain high coding standards across the project.

### CODE REVIEW

**Comment by BruhLemma-Yadecha** [comment]
If the line lengths look fine, I'll send a commit that fixes the ones with errors.

File: .flake8 [hunk 1]
```diff
@@ -0,0 +1,4 @@
+[flake8]
+max-line-length = 100
```
Comment (amhajja) [inline .flake8:2]: This should be 120

File: pyproject.toml [hunk 2]
```diff
@@ -0,0 +1,3 @@
+[tool.black]
+line-length = 100
```
Comment (amhajja) [inline pyproject.toml:2]: We can make this 120 also

**Review by amhajja — CHANGES_REQUESTED** [review]
Minor comments; the Black test will fail regardless because we're not changing Python files, but flake8 should pass.

**Pull request merged**